src/algebra/big_operators/basic.lean

```diff
@@ -349,37 +349,23 @@ lemma prod_sdiff [decidable_eq α] (h : s₁ ⊆ s₂) :
 by rw [←prod_union sdiff_disjoint, sdiff_union_of_subset h]
 
 @[simp, to_additive]
-lemma prod_sum_elim [decidable_eq (α ⊕ γ)]
-  (s : finset α) (t : finset γ) (f : α → β) (g : γ → β) :
-  ∏ x in s.map function.embedding.inl ∪ t.map function.embedding.inr, sum.elim f g x =
-    (∏ x in s, f x) * (∏ x in t, g x) :=
+lemma prod_sum_elim (s : finset α) (t : finset γ) (f : α → β) (g : γ → β) :
+  ∏ x in (s.map function.embedding.inl).disj_union (t.map function.embedding.inr)
+    (finset.disjoint_map_inl_map_inr _ _), sum.elim f g x =
+      (∏ x in s, f x) * (∏ x in t, g x) :=
```

Comment:
```suggestion
      (∏ x in s, f x) * ∏ x in t, g x :=
```

Reply:
That seems much worse to me. It's not against the rules to add parentheses that Lean doesn't need, especially when they add symmetry.

Reply:
Fine